### DISCUSSION TITLE
Is there a way to force analytics to send immediately?

### DISCUSSION THREAD

**Discussion by unikitty37** [Homebrew/discussions#4665] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3b3300546b5a4e40b74f4ee33cf225cca280defe
Last commit: 7 hours ago
Core tap JSON: 24 Jul 14:05 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BROWSER: /usr/bin/open
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vi
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.41.0 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: N/A
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I've just updated Homebrew and re-enabled analytics, now that it doesn't use Google. However, I run Little Snitch, so I would like to trigger an analytics send so I can approve it and forget about it, rather than have some overnight script invoke Homebrew and leave Homebrew hanging because it's tried to send analytics and is now blocked until I get back to the keyboard.

Is there a way to do this? Enabling analytics doesn't seem to immediately send an analytics event, unfortunately. (Obviously, this is only going to be a one-off for any Little Snitch user unless they change the duration from "forever", but it would still be nice to get it all out of the way right at the start :)

**Comment by SMillerDev** [comment, marked as answer]
> However, I run Little Snitch, so I would like to trigger an analytics send so I can approve it and forget about it, 

Installing something will report the install to Homebrew, there are no other analytics (and never have been).
> rather than have some overnight script invoke Homebrew and leave Homebrew hanging because it's tried to send analytics and is now blocked until I get back to the keyboard.

Homebrew will not hang on analytics. If it can't send it just won't send.

**Reply by unikitty37** [reply]
Thanks — I thought it would report on install, so I tried uninstalling and then reinstalling pnpm, but I got no Little Snitch dialog.

I got the same result when installing a formula I hadn't installed before: `brew install --cask font-envy-code-r-nerd-font`.

Running an install command in Terminal rather than iTerm revealed the answer: I had allowed all connections from iTerm via curl — definitely not a good idea if you want to actually approve/reject on an individual basis :)
